specs/Print.md

```diff
@@ -0,0 +1,473 @@
+
+// This example shows the user a print dialog. If `printDialogKind` is COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW, opens a browser print preview dialog, otherwise opens a system print dialog.
+```cpp
+bool AppWindow::OpenPrintDialog(COREWEBVIEW2_PRINT_DIALOG_KIND printDialogKind)
+{
+     wil::com_ptr<ICoreWebView2_15> webView2_15;
+     CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+     CHECK_FEATURE_RETURN(webView2_15);
+     CHECK_FAILURE(webView2_15->Print(printDialogKind));
+     return true;
+}
+```
+
+```c#
+void OpenPrintDialog(object target, ExecutedRoutedEventArgs e)
+{
+     string printDialog = e.Parameter.ToString();
+     if (printDialog == "PrintPreview")
+     {
+         // Opens the browser print preview dialog.
+         webView.CoreWebView2.Print();
+     }
+     else
+     {
+         // Opens the system print dialog.
+         webView.CoreWebView2.Print(CoreWebView2PrintDialogKind.SystemPrint);
+     }
+}
+```
+
+## PrintWithSettings
+
+You can use `PrintWithSettings` method to print to a specific printer without a print dialog. You can programatically configure print settings using `CreatePrintSettings`, and call `PrintWithSettings` to print the current page.
+
+// This example prints the current page without a print dialog to the specified printer with the settings.
+
+```cpp
+bool AppWindow::PrintWithSettings()
+{
+    TextInputDialog dialog(
+        GetMainWindow(), L"Printer Name", L"Printer Name:",
+        L"Specify printer name as understood by the OS.", L"");
+    if (dialog.confirmed)
+    {
+        wil::com_ptr<ICoreWebView2_15> webView2_15;
+        CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+        CHECK_FEATURE_RETURN(webView2_15);
+
+        wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+        CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+        CHECK_FEATURE_RETURN(webviewEnvironment6);
+
+        wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+        CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+        wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+        CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+        CHECK_FEATURE_RETURN(printSettings2);
+
+        CHECK_FAILURE(printSettings2->put_Orientation(COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT));
+        CHECK_FAILURE(printSettings2->put_PageWidth(8.5));
+        CHECK_FAILURE(printSettings2->put_PageHeight(11));
+        CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(true));
+        CHECK_FAILURE(printSettings2->put_PagesPerSheet(1));
+        CHECK_FAILURE(printSettings2->put_QualityHorizontal(600));
+        CHECK_FAILURE(printSettings2->put_QualityVertical(600));
+        CHECK_FAILURE(printSettings2->put_PrinterName((dialog.input).c_str()));
+
+        CHECK_FAILURE(webView2_15->PrintWithSettings(
```

Comment:
I think this is the one that should be called `Print`, because it actually prints.

Reply:
In WinRT this should be called `Print` and use [overload]. In COM I think it `PrintWithSettings` is right.